pallets/collator-selection/src/lib.rs

```diff
@@ -196,7 +196,9 @@ pub mod pallet {
 	#[pallet::genesis_build]
 	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
 		fn build(&self) {
-			// TODO: prevent duplicate.
+
+			let duplicate_invulnerables = self.invulnerables.iter().map(|x| x).cloned().collect::<std::collections::BTreeSet<_>>();
+			assert!(duplicate_invulnerables.len()  == self.invulnerables.len(), "duplicate invulnerables in genesis.");
```

Comment:
I don't quite get this; how is this preventing dupes? 

If incorrect, you can `clone`, `dedup`, and `compare`. Alternatively, keep sorted and binary search each element.

Reply:
Ahh I am blind, the `BtreeSet`. 

The `.map` is redundant though, otherwise okay.